src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -149,26 +148,63 @@ await processManager.ExecuteGit(Path.GetDirectoryName(tmpProductRepo)!, [
     private async Task<IReadOnlyCollection<string>> GetDiffFilters(string mapping)
     {
         var gitRepo = gitRepoFactory.CreateClient(DarcLib.Constants.DefaultVmrUri);
-        var repo = await remoteFactory.CreateRemoteAsync(DarcLib.Constants.DefaultVmrUri);
         return sourceMappingParser.ParseMappingsFromJson(
             (await gitRepo.GetFileContentsAsync($"{VmrInfo.SourceDirName}/{VmrInfo.SourceMappingsFileName}", DarcLib.Constants.DefaultVmrUri, "main")))
             .First(m => m.Name == mapping).Exclude;
     }
 
+    private async Task<string> GetDefaultVmrRepoRef(string defaultRemote)
+    {
+        var gitRepo = gitRepoFactory.CreateClient(DarcLib.Constants.DefaultVmrUri);
+        return sourceMappingParser.ParseMappingsFromJson(
+            (await gitRepo.GetFileContentsAsync($"{VmrInfo.SourceDirName}/{VmrInfo.SourceMappingsFileName}", DarcLib.Constants.DefaultVmrUri, "main")))
+            .First(m => m.DefaultRemote == defaultRemote).DefaultRef;
+    }
+
     private async Task CheckoutBranchAsync(DiffRepo repo) =>
         await localGitRepoFactory.Create(new NativePath(repo.Path)).CheckoutAsync(repo.Branch);
 
     private async Task<DiffRepo> ParseRepo(string input)
     {
-        var branchIndex = input.LastIndexOf(':');
+        string repo, branch;
+        int searchStartIndex = 0;
+        bool isLocal, isVmr;
+        if (input.StartsWith(HttpsPrefix))
+        {
+            searchStartIndex = HttpsPrefix.Length;
+            isLocal = false;
+        }
+        else
+        {
+            isLocal = true;
+            if (char.IsLetter(input[0]) && input[1] == ':')
+            {
+                searchStartIndex = 2;
+            }
```

Comment:
What is this case for? Can we put a comment?

Reply:
it's for windows paths, things like C:/Users/User/Desktop.., will add a comment

Reply:
yeah, drives can have more than 1 char for example, but it should be good enough.. We can probably write some regex if we really need too